### ISSUE TITLE
refactor ```lr_scheduler```

### ISSUE THREAD

**Issue by whr94621** [whr94621/NJUNMT-pytorch#29]
The way we configured ```lr_scheduler``` before is a little bit messy:

- We should configure it under ```optimizer_configs``` as this part is actually related to optimization. Also there too many options under ```training_configs```
- Lack of a ```d_model``` option to ```noam``` scheduler
- Lack of docs

So I push a new branch about the my refactoring of this part. Feel free to add some comments below.
